src/plugins/data/public/ui/query_editor/language_selector.tsx

```diff
@@ -57,8 +56,8 @@ export const QueryLanguageSelector = (props: QueryLanguageSelectorProps) => {
 
   languageService.getLanguages().forEach((language) => {
     if (
-      (language && props.appName && !language.supportedAppNames.includes(props.appName)) ||
-      uiService.Settings.getUserQueryLanguageBlocklist().includes(language?.id)
+      (language && props.appName && !language.editorSupportedAppNames?.includes(props.appName)) ||
+      languageService.getUserQueryLanguageBlocklist().includes(language?.id)
```

Comment:
here is where we can probably modify this to ensure that the language service.getlanguages for each language only returns backs languages that are available from the dataset.

tldr: we should just getLanguages return back already filtered instead of the lagnuage selector handling this